internal/runner/pytest.go

```diff
@@ -156,3 +163,19 @@ func ParsePytestCollectorResult(path string) ([]TestEngineTest, error) {
 
 	return results, nil
 }
+
+func getRandomTempFilename() string {
+	max := int64(1e10)
+	n, err := rand.Int(rand.Reader, big.NewInt(max))
+	if err != nil {
+		panic(err)
+	}
+	return filepath.Join(os.TempDir(), fmt.Sprintf("pytest-results-%d.json", n))
+}
```

Comment:
not blocking, but alternatively we can use [os.MkdirTemp](https://pkg.go.dev/os#MkdirTemp) with `*` pattern to create unique directory so we don't have to create unique identifier ourselves.
```suggestion
func getRandomTempFilename() string {
	tempDir, err := os.MkdirTemp("", "pytest-collector-*")
	if err != nil {
		panic(err)
	}
	return filepath.Join(tempDir, "result.json")
}
```

Reply:
Good idea 👍🏿